python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,582 @@
+                index = index.to_pandas()
+        output = [geom.to_shapely() for geom in self]
+        return gpGeoSeries(output, index=index)
+
+    def __repr__(self):
+        return self.to_pandas().__repr__() + "\n" + "(GPU)" + "\n"
+
+    def _dump(self):
+        return (
+            "POINTS"
+            + "\n"
+            + self.points.__repr__()
+            + "MULTIPOINTS"
+            + "\n"
+            + self.multipoints.__repr__()
+            + "LINES"
+            + "\n"
+            + self.lines.__repr__()
+            + "POLYGONS"
+            + "\n"
+            + self.polygons.__repr__()
+            + "\n"
+        )
+
+    def de_interleave(self):
+        return NotImplementedError
+
+    def copy(self, deep=True):
+        result = GeoSeries([])
+        result._data = self._data
+        result._reader = self._reader
+        result._points = self.points.copy(deep)
+        result._multipoints = self.multipoints.copy(deep)
+        result._lines = self.lines.copy(deep)
+        result._polygons = self.polygons.copy(deep)
+        result.types = self.types
+        result.lengths = self.lengths
+        result.index = self.index
+        return result
+
+    def __iter__(self):
+        self._iter_index = 0
+        return self
+
+    def __next__(self):
+        if self._iter_index >= len(self):
+            raise StopIteration
+        result = self[self._iter_index]
+        self._iter_index = self._iter_index + 1
+        return result
+
+
+class GpuPointArray:
+    def __init__(self):
+        self.xy = None
+        self.z = None
+        self.has_z = False
+
+    def __iter__(self):
+        self._index = 0
+        return self
+
+    def __next__(self):
+        if self._index >= len(self.xy):
+            raise StopIteration
+        result = self[self._index]
+        self._index = self._index + 2
+        return result
+
+    def __getitem__(self, index):
+        if isinstance(index, slice):
+            new_index = slice(index.start * 2, index.stop * 2 + 1, index.step)
+            return self.xy.iloc[new_index]
+        return self.xy.iloc[(index * 2) : (index * 2) + 2]
+
+    def __repr__(self):
+        return "xy: " + self.xy.__repr__()
+
+    def copy(self, deep=True):
+        result = GpuPointArray()
+        if self.xy is not None:
+            result.xy = self.xy.copy(deep)
+        if self.z is not None:
+            result.z = self.z.copy(deep)
+        result.has_z = self.has_z
+        return result
+
+
+class GpuOffsetArray(GpuPointArray):
+    def __init__(self):
+        super().__init__()
+        self.offsets = []
+
+    def __iter__(self):
+        if self.offsets is None:
+            raise TypeError
+        self._index = 1
+        return self
+
+    def __next__(self):
+        if self._index >= len(self.offsets):
+            raise StopIteration
+        result = self[self._index]
+        self._index = self._index + 1
+        return result
```

Comment:
Let's chat about this. This method fills out the rows of the GeoSeries, primarily for converting back to `geopandas`.

Reply:
Ok, so the base `GeoColumn` object provides an iterator for each row of the Column - this is because each row is actually a composite object of one or more geometries, and those geometries are constructed by combining various offset values.

All the other iterators are removed.